src/main/java/de/teamlapen/vampirism/misc/SettingsProvider.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    public void syncSettingsCache() {
+        getSettingValuesAsync().handleAsync(this::checkSettings).thenAccept(x -> {
+            this.settingValues.clear();
+            this.settingValues.putAll(x);
+        });
+    }
+
+    @Nullable
+    public String getSettingsValue(String key) {
+        return this.settingValues.get(key);
+    }
+
+    public @NotNull Optional<String> getSettingsValueOpt(String key) {
+        return Optional.ofNullable(this.settingValues.get(key));
+    }
+
+    public boolean isSettingTrue(String key) {
+        return "true".equals(this.settingValues.get(key));
+    }
+
+    @Override
+    public @NotNull CompletableFuture<Collection<Supporter>> getSupportersAsync() {
+        return GetSupportersAsync().handleAsync(this::checkSupporter);
+    }
+
+    @Nullable
+    public CompletableFuture<String> getSettingValueAsync(String key) {
+        return get("config/get?configId=" + key);
+    }
+
+    public CompletableFuture<Map<String, String>> getSettingValuesAsync() {
+        return get("config/list").thenApply(x -> GSON.fromJson(x, TypeToken.getParameterized(Map.class, String.class, String.class).getType()));
+    }
+
+    public CompletableFuture<Map<String, String>> getSettingValuesAsync(String modid) {
+        return get("config/list?modid=" + modid).thenApply(x -> GSON.fromJson(x, TypeToken.getParameterized(Map.class, String.class, String.class).getType()));
+    }
+
+    public CompletableFuture<Collection<Supporter>> GetSupportersAsync() {
+        Type type = TypeToken.getParameterized(List.class, Supporter.class).getType();
+        return get("supporter/list").thenApply(x -> GSON.fromJson(x, TypeToken.getParameterized(List.class, Supporter.class).getType()));
+    }
+
+    public CompletableFuture<Collection<Supporter>> GetSupportersAsync(String modid) {
+        return GetSupportersAsync().thenApply(x -> x.stream().filter(y -> y.faction().getNamespace().equals(modid)).collect(Collectors.toList()));
+    }
+
+    private CompletableFuture<String> get(String path) {
+        return get(this.apiVersion, path);
+    }
+
+    private CompletableFuture<String> get(String apiVersion, String path) {
+        try {
+            var request = HttpRequest.newBuilder(new URI(this.baseUrl + "/" + apiVersion + "/" + path)).GET().build();
+            return this.client.sendAsync(request, HttpResponse.BodyHandlers.ofString()).thenApply(HttpResponse::body);
+        } catch (URISyntaxException e) {
+            return CompletableFuture.failedFuture(e);
+        }
+    }
+
+    private Map<String, String> checkSettings(Map<String, String> settings, Throwable error) {
+        if (error != null) {
+            LOGGER.error("Failed to retrieve settings from server", error);
+        }
+        if (VampirismMod.inDev || settings == null) {
+            try {
+                InputStream inputStream = VampirismMod.class.getResourceAsStream("/config_values.json");
```

Comment:
Maybe name file `default_remote_config`  or `fallback_api_values`. Also IOException text below is copy pasta